docs/website/versioned_docs/version-3.0.0/command-reference/delete.md

```diff
@@ -1,40 +1,39 @@
 ---
-title: odo delete
+title: odo delete component
 sidebar_position: 4
 ---
 
-`odo delete` command is useful for deleting resources that are managed by odo.
+`odo delete component` command is useful for deleting resources that are managed by odo. It deletes the component and it's related innerloop and outerloop resources from the cluster.
 
-## Deleting a component
-
-To delete a _devfile_ component, you can execute `odo delete`.
+There are 2 ways to deleting a component:
+1. [Delete with access to Devfile](/#Delete with access to Devfile)
+2. [Delete without access to Devfile](/#Delete without access to Devfile)
 
+## Delete with access to Devfile
 ```shell
-odo delete
+odo delete component
 ```
+odo analyzes the devfile for the devfile component and the outerloop resources.
+If the component has been deployed on the cluster, then odo will list all the resources and prompt the user to confirm the deletion.
+Otherwise, odo will exit with a message stating that it could not find the resources on the cluster.
 
-If the component is pushed to the cluster, running the above command will delete the component from the cluster, and it's dependant storage, url, secrets, and other resources.
-If it is not pushed, the command would exit with an error stating that it could not find the resources on the cluster.
-
-Use `-f` or `--force` flag to avoid the confirmation questions. 
-
-## Un-deploying Devfile Kubernetes components
+_Note:_ odo does not delete the Devfile, and config files.
 
-To undeploy the Devfile Kubernetes components deployed with `odo deploy` from the cluster, you can execute the `odo delete` command with `--deploy` flag:
+## Delete without access to Devfile
 ```shell
-odo delete --deploy
+odo delete component --name <component_name> --namespace <namespace>
 ```
 
-Use `-f` or `--force` flag to avoid the confirmation questions.
+odo builds a label from the component name to search for the component on the cluster in the given namespace, and deletes it along with all it's related resources.
```

Comment:
Do we need to share these details to a regular odo user? No, IMO. What we do need to cover in this and the section on deletion with devfile is that one command deletes only the resources it sees in devfile, while the other deletes all the resources based on the name of the component.

@kadel I think you agreed that `odo delete component --name <name>` is not a good way to differentiate between deletion with and without a devfile. Seems like it's already gone in. And, unfortunately, it was me who approved that PR (#5574).

Reply:
> odo delete component --name <name> is not a good way to differentiate between deletion with and without a devfile. Seems like it's already gone in. And, unfortunately, it was me who approved that PR (https://github.com/redhat-developer/odo/pull/5574).

This was established way before the cleanup PR went in.

Reply:
> @kadel I think you agreed that odo delete component --name <name> is not a good way to differentiate between deletion with and without a devfile. Seems like it's already gone in. And, unfortunately, it was me who approved that PR (https://github.com/redhat-developer/odo/pull/5574).

This is in now, at some point we all agreed on this approach.  We don't have a consensus on an alternative. So, for now, --name it is. We still can change it before the final v3 but it should e discussed in a separate issue.

Reply:
> Do we need to share these details to a regular odo user? No, IMO. What we do need to cover in this and the section on deletion with devfile is that one command deletes only the resources it sees in devfile, while the other deletes all the resources based on the name of the component.

Normal user probably don't need to know that, but I like the fact that this is mentioned in the docs.